src/iris_api/bin/sender.py

```diff
@@ -808,6 +808,14 @@ def distributed_send_message(message):
 
 def fetch_and_send_message():
     message = send_queue.get()
+
+    if not cache.application_quotas(message):
+        logger.warn('Hard message quota exceeded; Dropping this message on floor: %s', message)
+        message_id = message.get('message_id')
+        if message_id:
```

Comment:
We should also log if non persisted messages are being dropped. Also we should add a metric for this.